problemtools/verifyproblem.py

```diff
@@ -245,11 +215,32 @@ def __init__(self, problem, datadir, parent=None):
             except Exception as e:
                 self.error(e)
                 self.config = {}
-        elif parent is not None:
-            self.config = parent.config.copy()
         else:
             self.config = {}
 
+        # For non-root groups, missing properties are inherited from the parent group
+        if parent:
+            for field, parent_value in parent.config.iteritems():
+                if not field in self.config:
+                    self.config[field] = parent_value
+
+        # Some deprecated properties are inherited from problem config during a transition period
+        problem_grading = problem.config.get('grading')
+        for key in ['accept_score', 'reject_score', 'range']:
+            if key in problem.config.get('grading'):
+                self.config[key] = problem_grading[key]
+
+        problem_on_reject = problem_grading.get('on_reject')
+        if problem_on_reject == 'first_error':
+            self.config['on_reject'] = 'break'
+        if problem_on_reject == 'grade':
+            self.config['on_reject'] = 'continue'
```

Comment:
Yes, raising an error when both are specified is probably reasonable.  It may be a bit wonky to do that though, since you can't raise errors from the constructor.

But actually, I think I take it back, it's probably better to use the values from problem.yaml everywhere if they are specified, since that's the current behaviour.

In other words, ignore all of this comment.